src/test/java/io/reactivex/validators/TestPrefixInMethodName.java

```diff
@@ -0,0 +1,131 @@
+            return;
+        }
+
+        Queue<File> dirs = new ArrayDeque<File>();
+
+        StringBuilder fail = new StringBuilder();
+        fail.append("The following code pattern was found: ").append(pattern).append("\n");
+        fail.append("Refresh and re-run tests!\n\n");
+
+        File parent = f.getParentFile();
+
+        dirs.offer(new File(parent.getAbsolutePath().replace('\\', '/')));
+        dirs.offer(new File(parent.getAbsolutePath().replace('\\', '/').replace("src/main/java", "src/test/java")));
+
+        Pattern p = Pattern.compile(pattern);
+
+        int total = 0;
+
+        while (!dirs.isEmpty()) {
+            f = dirs.poll();
+
+            File[] list = f.listFiles();
+            if (list != null && list.length != 0) {
+
+                for (File u : list) {
+                    if (u.isDirectory()) {
+                        dirs.offer(u);
+                    } else {
+                        String fname = u.getName();
+                        if (fname.endsWith(".java")) {
+
+                            int lineNum = 0;
+                            List<String> lines = new ArrayList<String>();
+                            BufferedReader in = new BufferedReader(new FileReader(u));
+                            boolean found = false;
+                            try {
+                                for (; ; ) {
+                                    String line = in.readLine();
+                                    if (line == null) {
+                                        break;
+                                    }
+                                    lineNum++;
+
+                                    Matcher matcher = p.matcher(line);
+                                    if (!line.startsWith("//") && !line.startsWith("*") && matcher.find()) {
+                                        found = true;
+                                        fail
+                                                .append(fname)
+                                                .append("#L").append(lineNum)
+                                                .append("    ").append(line)
+                                                .append("\n");
+                                        total++;
+
+                                        int methodNameStartIndex = matcher.end() - 1;
+                                        char firstChar = Character.toLowerCase(line.charAt(methodNameStartIndex));
+
+                                        String newLine = matcher.replaceAll(replacement + firstChar);
+
+                                        lines.add(newLine);
+                                    } else {
+                                        lines.add(line);
+                                    }
+
+                                }
+                            } finally {
+                                in.close();
+                            }
+
+                            if (found && System.getenv("CI") == null) {
```

Comment:
I'd rather not have this run unconditionally. There could be cases this introduces conflicts or other compilation errors without a good way of undoing the changes. Please comment this part out for now.

Reply:
Updated